examples/scripts/client.py

```diff
@@ -0,0 +1,258 @@
+        queue_name_query: str = urljoin(self.queue_endpoint, "name", name, "lock")
+        return requests.put(queue_name_query).json()
+
+    def unlock_queue(self, name: str):
+        queue_name_query: str = urljoin(self.queue_endpoint, "name", name, "lock")
+        return requests.delete(queue_name_query).json()
+
+    def register_experiment(self, name: str) -> Dict[str, Any]:
+        experiment_registration_form = {"name": name}
+
+        response = requests.post(
+            self.experiment_endpoint,
+            data=experiment_registration_form,
+        )
+
+        return response.json()
+
+    def register_queue(self, name: str = "tensorflow_cpu") -> Dict[str, Any]:
+        queue_registration_form = {"name": name}
+
+        response = requests.post(
+            self.queue_endpoint,
+            data=queue_registration_form,
+        )
+
+        return response.json()
+
+    def submit_job(
+        self,
+        workflows_file: PathLike,
+        experiment_name: str,
+        entry_point: str,
+        entry_point_kwargs: Optional[str] = None,
+        depends_on: Optional[str] = None,
+        queue: str = "tensorflow_cpu",
+        timeout: str = "24h",
+    ) -> Dict[str, Any]:
+        job_form = {
+            "experiment_name": experiment_name,
+            "queue": queue,
+            "timeout": timeout,
+            "entry_point": entry_point,
+        }
+
+        if entry_point_kwargs is not None:
+            job_form["entry_point_kwargs"] = entry_point_kwargs
+
+        if depends_on is not None:
+            job_form["depends_on"] = depends_on
+
+        workflows_file = Path(workflows_file)
+
+        with workflows_file.open("rb") as f:
+            job_files = {"workflow": (workflows_file.name, f)}
+            response = requests.post(
+                self.job_endpoint,
+                data=job_form,
+                files=job_files,
+            )
+
+        return response.json()
+
+    def upload_custom_plugin_package(
+        self,
+        custom_plugin_name: str,
+        custom_plugin_file: PathLike,
+        collection: str = "dioptra_custom",
+    ) -> Dict[str, Any]:
+        plugin_upload_form = {
+            "task_plugin_name": custom_plugin_name,
+            "collection": collection,
+        }
+
+        custom_plugin_file = Path(custom_plugin_file)
+
+        with custom_plugin_file.open("rb") as f:
+            custom_plugin_file = {"task_plugin_file": (custom_plugin_file.name, f)}
+            response = requests.post(
+                self.task_plugin_endpoint,
+                data=plugin_upload_form,
+                files=custom_plugin_file,
+            )
+
+        return response.json()
+
+
+def tar_helper(
+    source_dir: list[Union[str, Path]],
+    tarball_filename: str,
+    tarball_write_mode: str = "w:gz",
+    working_dir: Optional[Union[str, Path]] = None,
+) -> None:
+    """Archives a list of directories and files and compresses the archive.
+    Args:
```

Comment:
Style comment:

Google doc-strings should have a blank line separating each section.